com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Project.java

```diff
@@ -1773,32 +1782,40 @@ private List<TaskResult> doBuild(IProgress monitor, String... commands) throws T
                         }
                     }
                     TimeProfiler.stop();
-
-                    if (shouldBuildProject) {
-                        result = createAndRunTasks(monitor);
+                    boolean resourceBuildingFailed = false;
+                    try {
+                        if (shouldBuildProject) {
+                            result = createAndRunTasks(monitor);
+                        }
                     }
-
-                    if (remoteBuildFuture != null) {
-                        // get the result from the remote build and catch
-                        // if an exception was thrown in buildRemoteEngine() the
-                        // original exception is included in the ExecutionException
-                        try {
-                            remoteBuildFuture.get();
+                    catch(Exception e) {
+                        if (!monitor.isCanceled()) {
+                            resourceBuildingFailed = true;
+                            throw e;
                         }
-                        catch (ExecutionException|InterruptedException e) {
-                            Throwable cause = e.getCause();
-                            if ((cause instanceof MultipleCompileException) ||
-                                (cause instanceof CompileExceptionError)) {
-                                throw cause;
+                    }
+                    finally {
```

Comment:
If build canceled and canceled because of engine build fails, we need to throw this exception here